torch/csrc/jit/tensorexpr/llvm_codegen.cpp

```diff
@@ -1066,13 +1077,29 @@ void LLVMCodeGen::visit(const Intrinsics* v) {
     }
   }
 
+  std::unordered_set<IntrinsicsOp> simd_unary_intrinsics = {kErf,
+                                                            kErfc,
+                                                            kTan,
+                                                            kAcos,
+                                                            kAsin,
+                                                            kAtan,
+                                                            kCosh,
+                                                            kSinh,
+                                                            kTanh,
+                                                            kExpm1,
+                                                            kLgamma};
+
   std::vector<llvm::Value*> params;
   for (auto& p : v->params()) {
     p->accept(this);
     params.push_back(value_);
   }
 
-  if (v->dtype().lanes() == 1) {
+  if (v->dtype().lanes() == 1 ||
```

Comment:
Rather than having this super-complex if condition, how about having `bool scalarize` that is set / unset in the switch statement above.